test/optimism/LyraHandlerL2.ts

```diff
@@ -0,0 +1,309 @@
+  it("Contract initialized correctly", async function () {
+    expect(await lyraL2Handler.keeper()).equals(
+      signer.address,
+      "signer address check"
+    );
+    expect(await lyraL2Handler.wantTokenL2()).equals(usdc.address);
+    expect(await lyraL2Handler.lyraOptionMarket()).equals(
+      testSystem.optionMarket.address
+    );
+    expect(await lyraL2Handler.positionHandlerL1()).equals(signer.address);
+    expect(await lyraL2Handler.socketRegistry()).equals(movrRegistry);
+
+    const usdcBalance = await usdc.balanceOf(lyraL2Handler.address);
+    console.log("balance:", usdcBalance);
+
+    expect(usdcBalance.gt(0));
+  });
+
+  // Operation - Expected Behaviour
+  // deposit - Convert the usdc received to susd for purchasing option on lyra.
+  it("Deposit", async function () {
+    const beforeSusdBalance = await sUSD.balanceOf(lyraL2Handler.address);
+
+    await usdc
+      .connect(signer)
+      .transfer(lyraL2Handler.address, BigNumber.from(3000).mul(1e6));
+    const [startPositionValue] = await lyraL2Handler.positionInWantToken();
+
+    await lyraL2Handler.connect(signer).deposit();
+
+    const afterSusdBalance = await sUSD.balanceOf(lyraL2Handler.address);
+    expect(afterSusdBalance.gt(beforeSusdBalance)).equals(true);
+
+    const [finalPositionValue] = await lyraL2Handler.positionInWantToken();
+    const percentage = startPositionValue
+      .sub(finalPositionValue)
+      .mul(100)
+      .mul(1e4)
+      .div(startPositionValue)
+      .div(1e4);
+
+    console.log(
+      "[start pos, final pos]",
+      startPositionValue.toString(),
+      finalPositionValue.toString()
+    );
+    console.log("[deposit] percentage", percentage.toString());
+
+    expect(percentage.lt(BigNumber.from(10))).equals(true);
+  });
+
+  // // Operation - Expected Behaviour
+  // // openPosition - Purchase new call option on LyraOptionMarket
+  // //              - Should only work if sent using Keeper address
+  // //              - Lyra's accountBalance contract should reflect change in positionValue of our contract
+  // //              - Shouldnt work if LyraPosition is already active
+  it("Can purchase Call and sell Call", async function () {
+    expect((await lyraL2Handler.currentPosition()).isActive).equals(false);
+    const listingId = BigNumber.from(1);
+    const beforeBalance = await sUSD.balanceOf(lyraL2Handler.address);
+    console.log("SUSD Balance:", beforeBalance.toString());
+
+    const optimalAmount = ethers.utils.parseEther("10");
+
+    console.log("[call] optimal amount to buy:", optimalAmount.toString());
+
+    await lyraL2Handler
+      .connect(signer)
+      .openPosition(listingId, true, optimalAmount, false);
+
+    console.log(
+      "test quote asset balance:",
+      await testSystem.snx.quoteAsset.balanceOf(lyraL2Handler.address)
+    );
+
+    // expect((await lyraL2Handler.currentPosition()).isActive).equals(true);
```

Comment:
Add tests to validate if position params are correct. Also check if the value from `positionInWantToken` matches up. Need to validate this even without the oracle once.